pkg/streamingpromql/operators/test_operator.go

```diff
@@ -26,7 +28,15 @@ func (t *TestOperator) ExpressionPosition() posrange.PositionRange {
 }
 
 func (t *TestOperator) SeriesMetadata(_ context.Context) ([]types.SeriesMetadata, error) {
-	return testutils.LabelsToSeriesMetadata(t.Series), nil
+	series := testutils.LabelsToSeriesMetadata(t.Series)
+	seriesPool, err := types.SeriesMetadataSlicePool.Get(len(series), t.MemoryConsumptionTracker)
+	if err != nil {
+		return nil, err
+	}
+
+	seriesPool = seriesPool[:len(series)]
+	copy(seriesPool, series)
```

Comment:
The use of `copy` here seems a bit awkward - perhaps we should reproduce the behaviour of `LabelsToSeriesMetadata` in this method so we don't need the copy?

Reply:
Updated in afcae368b0fcd074d6b95ab57a37aca95718d06f